staging/src/k8s.io/apiserver/pkg/endpoints/filters/audit.go

```diff
@@ -160,3 +171,63 @@ func decorateResponseWriter(responseWriter http.ResponseWriter, out io.Writer, i
 	}
 	return delegate
 }
+
+var _ http.ResponseWriter = &auditResponseWriter{}
+
+// auditResponseWriter intercepts WriteHeader, sets it in the event. If the sink is set, it will
+// create immediately an event (for long running requests).
+type auditResponseWriter struct {
+	http.ResponseWriter
+	event *auditinternal.Event
+	once  sync.Once
+	sink  audit.Sink
+}
+
+func (a *auditResponseWriter) processCode(code int) {
+	a.once.Do(func() {
+		if a.sink != nil {
+			a.sink.ProcessEvent(a.event)
+		}
+
+		// for now we use the ResponseStatus as marker that it's the first or second event
+		// of a long running request. As soon as we have such a field in the event, we can
+		// change this.
+		if a.event.ResponseStatus == nil {
+			a.event.ResponseStatus = &metav1.Status{}
+		}
+		a.event.ResponseStatus.Code = int32(code)
```

Comment:
It could be technically (if you write your custom `Status`), but probably isn't in the current code. Nobody knows what a user provided apiserver does.